### DISCUSSION TITLE
Subscription In Graphql API

### DISCUSSION THREAD

**Discussion by satyam120900** [graphql-dotnet/graphql-dotnet#3892] [Q&A] [answered]
Hi Anyone Can Share the code for making subscription in graphql in dot net using graphql.server.transport.aspnetcore
and also tell the Configuration that need to be done in Program.cs and nugget packages to install

**Comment by Shane32** [comment, marked as answer]
There is extensive documentation in the readme file here for setting up the server, and typically no additional configuration is necessary to enable subscriptions:

https://github.com/graphql-dotnet/server

Near the bottom is a list of sample projects:

https://github.com/graphql-dotnet/server?tab=readme-ov-file#samples

Source code is here:

https://github.com/graphql-dotnet/server/tree/master/samples

Most of the samples include a working GraphQL subscription functionality.  I would suggest starting with the 'Basic' sample.

These samples support subscriptions:
- Basic
- Complex
- Controller via GraphQLExecutionActionResult
- Cors
- EndpointRouting
- Jwt
- MultipleSchemas
- Net48
- Pages

These samples do not include subscription functionality:
- Authorization
- Controller via ExecutionResultActionResult

GraphQL.NET has some limited documentation on subscriptions here:

https://graphql-dotnet.github.io/docs/getting-started/subscriptions/

Please ask if you have more specific questions.

**Reply by satyam120900** [reply]
Hi Shane Can you tell Which Nugget packages I need to install to achieve the Subscription. I was following the above document but I was unable to create subscription

**Reply by Shane32** [reply]
You only need [GraphQL.Server.Transports.AspNetCore](https://www.nuget.org/packages/GraphQL.Server.Transports.AspNetCore) which will include an indirect reference to [GraphQL](https://www.nuget.org/packages/GraphQL) and other required packages.  However it is likely you will want [GraphQL.Server.Ui.Playground](https://www.nuget.org/packages/GraphQL.Server.Ui.Playground) and [System.Reactive](https://www.nuget.org/packages/System.Reactive) as well.

**Reply by Shane32** [reply]
I suggest copying the Basic sample and its schema from here:

https://github.com/graphql-dotnet/server/tree/master/samples/Samples.Basic

https://github.com/graphql-dotnet/server/tree/master/samples/Samples.Schemas.Chat

They should run on your computer after fixing the package references.  Once it's running on your PC you can then swap the schema to a code-first schema as demonstrated in the documentation link.

**Reply by satyam120900** [reply]
Thanks @Shane32 It worked. I was very much confused with the Nugget package but finally It works Thanks once again.
